compositor_common/src/scene.rs

```diff
@@ -1,41 +1,83 @@
 use serde::{Deserialize, Serialize};
-use std::{any::Any, collections::HashMap, sync::Arc};
+use std::{collections::HashMap, sync::Arc};
 
-#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
-pub struct VideoId(usize);
-
-#[derive(Debug, Clone, PartialEq, Eq, Hash)]
-pub struct TransformationRegistryKey(pub String);
+use crate::transformation::TransformationRegistryKey;
 
 #[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, Serialize, Deserialize)]
 pub struct Resolution {
     pub width: usize,
     pub height: usize,
 }
 
-#[derive(Debug, Clone, Hash, PartialEq, Eq)]
-pub struct NodeId(pub String);
+#[derive(Debug, Clone, Hash, PartialEq, Eq, Serialize, Deserialize)]
+pub struct NodeId(pub Arc<str>);
 
-#[derive(Debug)]
-pub enum Node {
-    Video {
-        id: VideoId,
-        resolution: Resolution,
-    },
+#[derive(Debug, Clone, Hash, PartialEq, Eq, Serialize, Deserialize)]
+pub struct InputId(pub NodeId);
 
-    Image {
-        data: Vec<u8>,
-        resolution: Resolution,
-    },
+#[derive(Debug, Clone, Hash, PartialEq, Eq, Serialize, Deserialize)]
+pub struct OutputId(pub NodeId);
+
+impl From<NodeId> for InputId {
+    fn from(value: NodeId) -> Self {
+        Self(value)
+    }
+}
+
+impl From<NodeId> for OutputId {
+    fn from(value: NodeId) -> Self {
+        Self(value)
+    }
+}
+
+/// SceneSpec represents configuration that can be used to create new Scene
+/// or update an existing one.
+#[derive(Serialize, Deserialize)]
+pub struct SceneSpec {
+    pub inputs: Vec<InputSpec>,
+    pub transforms: Vec<TransformNodeSpec>,
+    pub outputs: Vec<OutputSpec>,
+}
 
-    Transformer {
-        registry_key: TransformationRegistryKey,
-        inputs: HashMap<NodeId, Arc<Node>>,
-        resolution: Resolution,
-        params: Box<dyn Any>,
+#[derive(Serialize, Deserialize)]
+pub struct InputSpec {
+    pub input_id: InputId,
+    pub resolution: Resolution,
+}
```

Comment:
I think we decided against it at the last meeting, and we will make all transforms accept dynamic sizes.

Reply:
I meant, that we should accept both ways of specifying the resolution of an output of an input node. Either as a fixed value, which would cause rescaling of each received frame (probably during YUV -> RGBA render) or as a received stream resolution (dynamic resolution).

I get the impression, that we want to support both options, but we can clarify it.

Reply:
I disagree. If we are going to support dynamic size, then I don't think we should support anything on the input level. If someone wants to ensure a fixed input size then they should have a transform that does that.

The input options that we discussed only make sense if we can't support dynamic size, but we can talk about it more on standup tomorrow.

Reply:
Ok, I get your point. 

Shaders usually don't need fixed-sized inputs (in most cases only ratio should matter, which should be changed with transform anyway), so I'm ok with using dynamic resolution for inputs, but we can clarify tomorrow that we all are on the same page.